etcdctl/READMEv3.md

```diff
@@ -735,6 +735,59 @@ cf1550fb, 3, 3, 25 kB
 +----------+----------+------------+------------+
 ```
 
+### MIGRATE [options]
+
+Migrate migrates keys in store v2 into mvcc store.
+
+#### Options
+
+- data-dir -- Path to the data directory
+
+- wal-dir -- Path to the WAL directory
+
+- transformer -- Path to the user-provided transformer program (default if not provided)
+
+#### Return value
+
+Simple reply
+
+- Exit code is zero when migration is finished successfully.
+
+- Error string if migration failed. Exit code is non-zero.
+
+#### Default transformer
+
+If user does not provide a transformer program, migrate command will use the default transformer. The default transformer transforms keys in store v2 into mvcc store according to the following Go program:
+
+```go
+func transform(n *storev2.Node) *mvccpb.KeyValue {
+	if n.Dir {
+		return nil
+	}
+	kv := &mvccpb.KeyValue{
+		Key:            []byte(n.Key),
+		Value:          []byte(n.Value),
+		CreateRevision: int64(n.CreatedIndex),
+		ModRevision:    int64(n.ModifiedIndex),
+		Version:        1,
+	}
+	return kv
+}
+```
+
+#### User-provided transformer
+
+Users can provide a customized transformer function that transforms a key from v2 store to any number of keys in mvcc store. Migration program writes JSON format of the [v2 store key][v2key] into the stdin of the transformer program, reads protobuf format of [mvcc key][v3key] back from the stdout of the transformer program, and save the transformed keys into mvcc store to finish migration.
```

Comment:
Users can provide a customized 1:n transformer function that transforms a key from the v2 store to any number of keys in the mvcc store. The migration program writes JSON formatted [v2 store keys][v2key] to the transformer program's stdin, reads protobuf formatted [mvcc keys][v3key] back from the transformer program's stdout, and finishes migration by saving the transformed keys into the mvcc store.